banzai_floyds/extract.py

```diff
@@ -189,81 +40,50 @@ def extract(binned_data):
     results = {'fluxraw': [], 'fluxrawerr': [], 'wavelength': [], 'binwidth': [], 'order': [], 'background': []}
     for data_to_sum in binned_data.groups:
         wavelength_bin = data_to_sum['wavelength_bin'][0]
+        # Skip pixels that don't fall into a bin we are going to extract
+        if wavelength_bin == 0:
+            continue
+        if data_to_sum['extraction_window'].sum() == 0:
+            continue
+        # Cut any bins that don't include the profile center. If the weights are small (i.e. we only caught the edge
+        # of the profile), this blows up numerically. The threshold here is a little arbitrary. It needs to be small
+        # enough to not have numerical artifacts but large enough to not reject broad profiles.
+        if np.max(data_to_sum['weights'][data_to_sum['extraction_window']]) < 5e-3:
+            continue
+
         wavelength_bin_width = data_to_sum['wavelength_bin_width'][0]
         order_id = data_to_sum['order'][0]
         # This should be equivalent to Horne 1986 optimal extraction
         flux = data_to_sum['data'] - data_to_sum['background']
         flux *= data_to_sum['weights']
         flux *= data_to_sum['uncertainty'] ** -2
-        flux = np.sum(flux)
-        flux_normalization = np.sum(data_to_sum['weights']**2 * data_to_sum['uncertainty']**-2)
+        flux = np.sum(flux[data_to_sum['extraction_window']])
+        flux_normalization = data_to_sum['weights']**2 * data_to_sum['uncertainty']**-2
+        flux_normalization = np.sum(flux_normalization[data_to_sum['extraction_window']])
         background = data_to_sum['background'] * data_to_sum['weights']
         background *= data_to_sum['uncertainty'] ** -2
-        background = np.sum(background)
+        background = np.sum(background[data_to_sum['extraction_window']])
         results['fluxraw'].append(flux / flux_normalization)
         results['background'].append(background / flux_normalization)
-        uncertainty = np.sqrt(np.sum(data_to_sum['weights']) / flux_normalization)
+        uncertainty = np.sqrt(np.sum(data_to_sum['weights'][data_to_sum['extraction_window']]) / flux_normalization)
         results['fluxrawerr'].append(uncertainty)
         results['wavelength'].append(wavelength_bin)
         results['binwidth'].append(wavelength_bin_width)
         results['order'].append(order_id)
     return Table(results)
 
 
-def combine_wavelegnth_bins(wavelength_bins):
```

Comment:
This logic was just wrong so I removed it. I will add an equivalent back in a future pr.